src/Forms/SearchableDropdownTrait.php

```diff
@@ -262,6 +262,10 @@ public function setIsSearchable(bool $isSearchable): static
      */
     public function getSource(): array
     {
+        // Source will be unused when lazy-loading so just return an empty array
+        if ($this->getIsLazyLoaded()) {
+            return [];
```

Comment:
This might result in unexpected behaviour down the line, but there's not a _lot_ we can do about that.

The one place that looks like this might cause problems is `MultiSelectField::performReadonlyTransformation()` - I think that'll need to be updated to use `$this->sourceList` instead of `$this->getSource()`.

Alternatively a new `getSourceList()` method could be added and used in most places that currently use `getSource()` - that's how TagField seems to have gotten around this problem.

Reply:
Yes unfortunately it does seem to break `MultiSelectField::performReadonlyTransformation()` when lazy loading

I don't think changing that to use $this->sourceList will really fix this issue, because ultimately it needs to be converted to use an array anyway because in readonly mode it's using a LookField, which internally uses arrays.

To fix this it feels like we need to convert a whole bunch of things to use generators to handle very large sets of options. Given there's a workaround to use a numeric field (which was the old fallback) feels like this is more effort than it's worth, at least for now.  Feels more like an API break piece of work where we also get rid of a bunch of the convoluted class hierarchy that exists for dropdowns

I'll close this PR

Reply:
The readonly list only needs to display the items from the list which _have actually been selected_ though, right? i.e. the current value of the field. Surely we can get a readonly formfield that displays the values without loading every single record in the DB into an array?

Reply:
Possibly

However looking at my closed PR in hindsight, it's honestly hackish. getSource() should just return the source and not care if it's lazyloaded or not